homeworks/veizerattila/4_exceptions_logging/to_do.py

```diff
@@ -0,0 +1,149 @@
+﻿"""Feladat 1: 
+Hozz létre egy to_do.py nevű file-t, és kódold le a következő feladat megoldását:
+- OK / Készíts egy Feladatkezelő alkalmazást! 
+- OK / Hozz létre egy .txt file-t és hagyd üresen.
+- OK / Definiálj 3 függvényt a következőkre: feladatok olvasása, egy feladat hozzáadása, egy feladat törlése.
+- OK / Legyen egy display_menu() function-öd is, ami kiprinteli a lehetséges opciókat:
+    * Add Task
+    * View Tasks
+    * Remove Task
+    * Exit
+- Folyamatosan kérj be inputot a felhasználótól hogy ezek közül a menüpontok közül mit szeretne csinálni, és hívd meg a válaszhoz megfelelő függvényt.
+- A felhasználó inputja 1,2,3 vagy 4 kell, hogy legyen, ellenőrizd! Ha az 1-es vagy 3-as opciót választja, mindkét esetben paramétert kell átadnod a megfelelő függvénynek. “Exit”-re lépjen ki a program.
+- Használj hibakezelést a file-ba való íráskor és olvasáskor, illetve használd a logging module-t. Egyszerre logolj a konzolra és egy .log file-ba. A .txt file legyen része a pull request-nek, de a log file-ok ne! Használd a .gitignore-t!
+
+Extra: Alakítsd át a .txt file-os megoldást úgy, hogy egy json file-t használsz a feladatok trackelésére!
+"""
+import time as tm
+import sys
+import os
+import logging
+
+filepath = "homeworks/veizerattila/4_exceptions_logging/To_do.txt"
+
+###############################################################################
+### Logolási környezet definiálása ############################################
+###############################################################################
+file_handler = logging.FileHandler("homeworks/veizerattila/4_exceptions_logging/To_do_app.log")
+stream_handler = logging.StreamHandler()
+
+formatter = logging.Formatter('%(asctime)s | %(levelname)s | %(message)s')
+
+file_handler.setFormatter(formatter)
+stream_handler.setFormatter(formatter)
+
+file_handler.setLevel(logging.INFO)
+stream_handler.setLevel(logging.INFO)
+
+logger = logging.getLogger()
+logger.addHandler(file_handler)
+logger.addHandler(stream_handler)
+logger.setLevel(logging.INFO)
+
+###############################################################################
+### Függvények definiálása ####################################################
+###############################################################################
+def view_task():
```

Comment:
Szia!

Ebben a function-ben nem szükséges raise-elni a FileNotFound Exception-t, vagy ha megteszed, akkor ahol hívod a function-t ott egy try-except-ben kéne lennie a hívásnak, hogy lekezeld a hibát.

Jelen helyzetben ennyi elég, egyszerűbb így:

```
def view_task():
    try:
        with open(filepath, "r", encoding='utf-8-sig') as file:
            logging.info("Viewing tasks.")
            for line in file:
                print(line.strip())
    except FileNotFoundError:
        logging.error(f"The file '{filepath}' does not exist.")
        print(f"The file '{filepath}' does not exist. Please create it or check the path.")
```

Reply:
Ezt is átvezettem, köszönöm a tippet.